antiplag/views.py

```diff
@@ -106,34 +117,44 @@ def get(self, request, id):
 
 
 class SubmissionGraphDetail(APIView):
-    def get(self, request, id):
-        submission = get_object_or_404(Submission, pk=id)
+    def get(self, request, submission_id):
+        submission = get_object_or_404(Submission, pk=submission_id)
 
-        nodes = []
+        nodes = {}
         links = []
         if submission.status == Submission.SubmissionStatus.PROCESSED:
             for doc in submission.documents.all():
                 doc_data = serializers.DocumentDetailedSerializer(doc).data
-                nodes.append(
-                    {"id": doc_data["id"], "name": doc_data["name"], "uploaded": True}
-                )
+                doc_id = doc_data["id"]
+                nodes[doc_id] = {
+                    "id": doc_id,
+                    "name": doc_data["name"],
+                    "uploaded": True,
+                }
 
                 matched_documents = doc.result.matched_docs
                 for matched_doc in matched_documents:
-                    nodes.append(
-                        {
+                    # It's either elastic document or our document
+                    matched_id = matched_doc.get("elastic_id", None) or matched_doc.get(
```

Comment:
Have you please tried it with frontend? Does this change involve any changes on frontend side?

Reply:
I haven't since it does not change resulting response. But I should have.

Reply:
Can you please test this properly before merging?